### PR TITLE
Remove Check Required w/ Tests

### LINKED ISSUE #43: Validate Without Requiring `require`
https://github.com/PolymerElements/iron-form/blob/master/iron-form.html#L243

The `if` only validates a custom element if it's `require`d, but fields may simply have a `pattern` or other validation.

### PR DESCRIPTION
**Pull request by patgmiller**
Validate all form input elements added tests for form validation.

Same as PR #125 which fixes #43 and adds tests for it.  

Please let me know if you want additional tests, I am happy to make them more rigorous if necessary. 
Thanks.

### CODE REVIEW

**Comment by patgmiller** [comment]
@notwaldorf have you had a chance to look at this since you requested some tests for the PR on validating non-required form inputs?

File: test/basic.html [hunk 1]
```diff
@@ -128,6 +128,22 @@
...
+      <form is="iron-form" action="/responds_with_json" method="post">
+        <paper-input label="numbers" pattern="^[0-9]$" ></paper-input>
+      </form>
+    </template>
+  </test-fixture>
+
+  <test-fixture id="FormWithPreventInvalidElements">
+    <template>
+      <form is="iron-form" action="/responds_with_json" method="post">
+        <input type="text" name="letters" pattern="^[a-z]$" >
```
Comment (notwaldorf) [inline test/basic.html]: `prevent-invalid` is a specific attribute on `iron-input`/`paper-input`, and i don't see it there. Maybe you just mean 'form with invalid pattern inputs' ?

**Comment by notwaldorf** [comment]
The code looks great, I just think there's a problem with the name of the fixtures -- `prevent-invalid-input` is a very specific attribute that you're not actually testing, so i think that's a deceiving name :)

**Comment by notwaldorf** [comment]
(i can merge this once you fix the name)

**Comment by notwaldorf** [comment]
Thanks so much for sticking through with this!

**Comment by mercmobily** [comment]
Totally missed, sorry! When will Polymer 1.7 be released? (Including this fix, I assume)
Thanks!

**Comment by patgmiller** [comment]
@mercmobily you don't need to wait for polymer 1.7 to be released or any new release of polymer to use this patch, bower supports installing from github directly, check out the [bower api - install](https://bower.io/docs/api/#install).

Also, each polymer element had its own release cycle, as soon as it has a new release tag bower will include that as a version it will install.

**Comment by mercmobily** [comment]
Ah silly me, very true! I should have written "When will the new version of this element be released?". and you would STILL answer my question -- I didn't know bower supported Githib that way! thanks...

**Comment by patgmiller** [comment]
You're welcome, as to release timing I can't say. I know @notwaldorf works on a lot of the polymer elements, so what I do when I am waiting is change my bower dependencies to use the specific branch or commit and install that while I am waiting on the new release.

**Comment by notwaldorf** [comment]
oh sorry! doing a release now :)

**Comment by notwaldorf** [comment]
@mercmobily Done! 1.1.0 is out. Sorry about the delay!

**Comment by patgmiller** [comment]
Thanks, no worries

**Pull request merged**